test/meson.build

```diff
@@ -16,19 +16,28 @@ endif
 
 c_openmp_simd = false
 cxx_openmp_simd = false
-foreach omp_arg : ['-fopenmp-simd', '-qopenmp-simd']
-  if (not c_openmp_simd) and cc.has_argument(omp_arg)
-    simde_c_args += omp_arg
-    simde_c_defs += '-DSIMDE_ENABLE_OPENMP'
-    c_openmp_simd = true
-  endif
 
-  if (not cxx_openmp_simd) and cxx.has_argument(omp_arg)
-    simde_cxx_args += omp_arg
-    simde_cxx_defs += '-DSIMDE_ENABLE_OPENMP'
-    cxx_openmp_simd = true
-  endif
-endforeach
+# LCC has OpenMP enabled by default without any args
+if cc.get_id() == 'lcc'
+  c_openmp_simd = true
+  cxx_openmp_simd = true
+  simde_c_defs += '-DSIMDE_ENABLE_OPENMP'
+  simde_cxx_defs += '-DSIMDE_ENABLE_OPENMP'
+else
+  foreach omp_arg : ['-fopenmp-simd', '-qopenmp-simd']
+    if (not c_openmp_simd) and cc.has_argument(omp_arg)
+      simde_c_args += omp_arg
+      simde_c_defs += '-DSIMDE_ENABLE_OPENMP'
+      c_openmp_simd = true
+    endif
+
+    if (not cxx_openmp_simd) and cxx.has_argument(omp_arg)
+      simde_cxx_args += omp_arg
+      simde_cxx_defs += '-DSIMDE_ENABLE_OPENMP'
+      cxx_openmp_simd = true
+    endif
+  endforeach
+endif
```

Comment:
If OpenMP SIMD support is always enabled, we should change [the checks for SIMDE_ENABLE_OPENMP](https://github.com/simd-everywhere/simde/blob/62c1087fa02e56cc69746e6ce4d26a9721e33ad0/simde/simde-common.h#L294). Just adding ` || HEDLEY_MCST_LCC_VERSION_CHECK(1,25,10)` should do the trick.

Reply:
So you enable OpenMP unconditionally with these checks? May it be the situation where we want to turn it off despite the compiler automatically turns it on?

Reply:
Yep.  I haven't run into a situation where we really want to disable OpenMP SIMD (except perhaps testing), but if that does happen IMHO it would be better to do something like

```
#if !defined(SIMDE_DISABLE_OPENMP)
  #if openmp_checks_here...
    #define SIMDE_ENABLE_OPENMP
  #endif
#endif
```

Instead of forcing the user to pass `-DSIMDE_ENABLE_OPENMP` (or doing it in code), that way we default to faster code. In general SIMDe is designed to just be dropped into a project without relying on the build system, that way any build system works, so choosing sensible defaults is important.

I know a lot of people don't like configuration magic in libraries, and I think that's a reasonable argument for traditional libraries, but they can safely hide that from the user.  With SIMDe all the code is in public headers, so that makes less sense… if we did that then *all* that logic would have to go in the build systems using SIMDe and be a huge maintenance burden ☹

Reply:
Well, that seems to be a good idea. So am I get it correct: we try to implicitly turn OpenMP SIMD on while building tests for SIMDe, if it is available; and then let SIMDe decide by itself whether to use it (if `_OPENMP` is defined, but `SIMDE_DISABLE_OPENMP` is not) or not, right? And thus we get rid of passing this `-DSIMDE_ENABLE_OPENMP` in `test/meson.build`, but force user to explicitly pass `-DSIMDE_DISABLE_OPENMP` in the rare case when they want to disable it.

Btw, don't we want to build tests for these two cases? Or just testing as it is now would be sufficient?

Reply:
Ok. So we build tests with OpenMP SIMD if we have its support, or OpenMP library instead, if it's available, or without OpenMP/OpenMP SIMD support, if there's nothing we can do about it.
And I went from explicit enabling of OpenMP SIMD by `-DSIMDE_ENABLE_OPENMP` to explicit disabling it by `-DSIMDE_DISABLE_OPENMP`. So, unless we specify `-DSIMDE_DISABLE_OPENMP`, we'll use OpenMP SIMD.